### PR TITLE
[Rev 2] Fixing jest tests, initial CI pipeline, adding commitizen, starter config for automation tools

### PR DESCRIPTION
**Pull request by aorumbayev**
### Description

Hey @drichar, this is a PR based on the older [PR](https://github.com/TxnLab/use-wallet/pull/31) but with merge conflicts fixed. Let me know if there are any questions, so far i just re applied all changes i had on the old pr

> _Please explain the changes you made here_

- Fixing jest tests. The tests weren't compiling correctly with typescript setup. Added a few changes to the Example page test and one unit test with snapshots.
- Custom unofficial logo for the project. I tried to get a font close enough to the NFD logo. If you'll like it i can also supply psd file separately so you can tweak it further yourself if needed in future 👍 
- Initial row of readme badges. You guys can later add ci status badges, coverage and etc if you want.
- Added commitizen and cz-conventional-changelog to help with commit messages. I can remove this if you guys don't want it. But it just aids with conventional commit messages. Simply run `yarn commit` and it will prompt you for the commit message. It can also generate the changelog for you. Task is completely optional.
- Adding pull request template. You can edit this to your liking. I just added a few things that I thought would be helpful.
- Adding yaml for auto_assign bot. This will automatically assign the PR to the project owner. If you want to use it after merging this PR, you will need to install the bot from the marketplace. Otherwise can remove this file just let me know.
- Adding renovate.json. This will help with keeping dependencies up to date via automated bot (ive set it to once a month with preference over aggregating minor bumps into single pr, then major bumps are always isolating individual packages). If you want to use it after merging this PR, you will need to install the bot from the marketplace. Otherwise can remove this file just let me know.
- Adding initial build pipeline. This will run the build and test tasks on every push or on PR to main.

### Checklist

> _Please, make sure to comply with the checklist below before expecting review_

- [x] Code compiles correctly
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by aorumbayev** [comment]
@drichar i see you are already using dependabot, if you want me to remove renovate just let me know, although i think its more convenient. Check out the example dependency dashboard it generates for the repo from which you can control dep updates https://github.com/AlgoWorldNFT/algoworld-swapper/issues/1

File: .gitignore
Comment (drichar) [inline .gitignore:20]: We'll want to keep this

File: README.md
Comment (drichar) [inline README.md:1]: Let's hold off on the logo for now. I tend to be very particular about branding and have some ideas for the logo that are still being fleshed out ;-)

I do like the extended description you provide below.

**Review by drichar — COMMENTED** [review]
Thank you for this @aorumbayev 

Fixing the Jest tests w/ TS is a huge help, once this is merged I'll start incrementally adding unit tests (overdue tbh). 

The initial CI pipeline, Commitizen and CommitLint, the PR template, all of it will amount to a much better DX. I haven't used Renovate before but it looks great. I'll set up the bot to replace Dependabot, along with the auto-assign bot.

After a few minor changes (see comments), I'll be very pleased to get this in 🙌

File: README.md
Comment (aorumbayev) [reply README.md:1]: https://ipfs.algonode.xyz/ipfs/bafkreiexg66kttj75i6uv6khowp2ybvue4nwhvuhh24dobpniszqkbuxoq

here is the image for reference - removed it from readme and replaced with simple textual description

File: .gitignore
Comment (aorumbayev) [reply .gitignore]: done

File: renovate.json
Comment (aorumbayev) [reply renovate.json:1]: You can always manually trigger an update pr from dependency dashboard if needed, been dealing with it for some time and smaller frequency can get very chatty and annoying, but regardless once merged feel free to tweak however you like ;)

**Review by drichar — APPROVED** [review]

**Pull request merged**
